pypit/core/arprocimg.py

```diff
@@ -768,10 +768,12 @@ def variance_frame(datasec_img, det, sciframe, idx, settings_det,
         varframe = np.abs(skyframe + objframe - np.sqrt(2)*np.sqrt(rnoise)) + rnoise
         return varframe
     else:
-        scicopy = sciframe.copy()
         # Dark Current noise
-        dnoise = (settings_det['darkcurr'] * float(fitsdict["exptime"][idx])/3600.0)
+        if dnoise is None:
+            dnoise = (settings_det['darkcurr'] * float(fitsdict["exptime"][idx])/3600.0)
         # Return
-        return np.abs(scicopy) + rnoise + dnoise
+        varframe = np.abs(sciframe - np.sqrt(2)*np.sqrt(rnoise)) + rnoise + dnoise
```

Comment:
This doesn't make sense to me. what is the factor `- np.sqrt(2)*np.sqrt(rnoise)` doing? I know that this is present in HIRESredux, but I didn't really understand it there either. I think it would be good to add a comment why this is needed.

Reply:
It is needed in the regime when there are very few counts.
It is not a precise estimate, but was a recipe we settled on 
after some experimentation.  Perhaps @jhennawi can recall
better the origin of this actual implementation.

Reply:
This is a hack to give approximately the correct variance in the low count limit where the gaussian approximation fails.

Reply:
Seems like we’re just shifting the zero point to be centred on sqrt(2*rnoise). I would think that a better approach might be to clip sciframe to have no values less than 1, and avoid taking the absolute value.

I appreciate that the implementation used in IDL has been well-tested, but it does seem a little arbitrary...

Reply:
Everything we do in this reduction pipeline assumes variances add in quadrature. That won't be the case for Poissonian noise. That means extracted noise will be wrong and weights will be wrong if you are truly in the Poisson limit.  This kludge of the inverse variance basically gives you approximately the variance of the Poisson distribution, which is probably the best thing you can do. It is also basically setting a floor on the inverse variance so that you don't give infinite weight to low-count pixels, and instead give them uniform weights.

Reply:
OK - thanks for the explanation - we should include this statement (or something like this) in a comment in the code, saying explicitly that the factor BLAH is a kludge. However, it still doesn't seem right to me... what happens in the case where `rnoise=2`? If you have `sciframe = 2`, then this term goes to zero and you're left with `varframe=rnoise + dnoise`. Note, this is also in the Poisson regime of low counts. So, unless I'm missing something, we should include an explanation in the code and/or a reference to where/how this kludge factor was determined (was it instrument specific?).

I guess I'm just pointing out that, I agree that `np.abs(scicopy) + rnoise + dnoise` is probably not the whole story, but the newly proposed equation is no better. So, we either need to justify one or the other, or come up with a new strategy.

Reply:
Ok, I've returned it to where it was and started Issue #354 
as I wish to merge this PR now.